src/lib/array.jl

```diff
@@ -341,3 +342,53 @@ end
 @adjoint function +(A::AbstractMatrix, S::UniformScaling)
   return A + S, Δ->(Δ, (λ=sum(view(Δ, diagind(Δ))),))
 end
+
+
+
+# FFTW
+# ===================
+
+# FFTW functions do not work with FillArrays, which are needed
+# for some functionality of Zygote. To make it work with FillArrays
+# as well, overload the relevant functions
+FFTW.fft(x::Fill, dims...) = FFTW.fft(collect(x), dims...)
+FFTW.ifft(x::Fill, dims...) = FFTW.ifft(collect(x), dims...)
+
+
+# the adjoint jacobian of an FFT with respect to its input is the reverse FFT of the
+# gradient of its inputs, but with different normalization factor
+@adjoint function FFTW.fft(xs)
+  return FFTW.fft(xs), function(Δ)
+      N = length(xs)
+      return (N * FFTW.ifft(Δ),)
+  end
+end
+
+@adjoint function FFTW.ifft(xs)
+  return FFTW.ifft(xs), function(Δ)
+      N = length(xs)
+      return (1/N* FFTW.fft(Δ),)
+  end
+end
+
+@adjoint function FFTW.fft(xs, dims)
+  return FFTW.fft(xs, dims), function(Δ)
+  # dims can be int, array or tuple,
```

Comment:
nit: indent these few lines?

Reply:
I (hopefully) fixed the indentation now.